src/PythonClient/src/quixstreams/dataframes/kafka/producer.py

```diff
@@ -0,0 +1,181 @@
+import asyncio
+import concurrent.futures
+import logging
+from typing import List, Dict, Literal, Tuple
+from typing import Union, Optional, Callable
+
+from confluent_kafka import KafkaException, Producer, KafkaError, Message
+
+from .exceptions import AsyncProducerNotStartedError
+
+__all__ = (
+    "AsyncProducer",
+    "Partitioner",
+)
+
+
+Partitioner = Literal[
+    "random", "consistent_random", "murmur2", "murmur2_random", "fnv1a", "fnv1a_random"
+]
+HeaderValue = Union[str | bytes | None]
+Headers = Union[
+    List[Tuple[str, HeaderValue]],
+    Dict[str, HeaderValue],
+]
+
+logger = logging.getLogger(__name__)
+confluent_logger = logging.getLogger("confluent_kafka.producer")
+
+
+def _default_error_cb(error: KafkaError):
+    logger.error(
+        "Kafka producer error",
+        extra={"error_code": error.code(), "error_desc": error.str()},
+    )
+
+
+class AsyncProducer:
```

Comment:
I don't see anything specific to SDF in this class so I would move it outside dataframes folder.

Reply:
Well, I disagree. There is still a use case for just reading events directly. I did not plan to hide this API from users.